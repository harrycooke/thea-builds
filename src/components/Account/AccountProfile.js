import React from 'react';
import PropTypes from 'prop-types';
import clsx from 'clsx';
import moment from 'moment';
import Dropzone from 'react-dropzone';
import { makeStyles } from '@material-ui/styles';
import {
  Card,
  CardHeader,
  CardActions,
  CardContent,
  Avatar,
  Typography,
  Divider,
  Button,
  //LinearProgress,
  Modal,
  Fab,
  CircularProgress,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
  Slide
} from '@material-ui/core';
import CloseIcon from "@material-ui/icons/Close";
import CloudDownloadIcon from "@material-ui/icons/CloudDownload";
import _ from "../../@lodash";
import { uploadFile, removeFileFromS3 } from '../../utils/s3';

const useStyles = makeStyles(theme => ({
  root: {},
  details: {
    display: 'flex',
    flexDirection: "column",
    margin: "1.5em"
  },
  avatar: {
    marginBottom: 20,
    height: 130,
    width: 130,
    flexShrink: 10,
    flexGrow: 0
  },
  progress: {
    marginTop: 20
  },
  uploadButton: {
    marginRight: 15,
    [theme.breakpoints.down('sm')]: {
      fontSize: 12  
    },
    [theme.breakpoints.up('md')]: {
      fontSize: 15
    },
  },
  modalContainer: {
    minWidth: 450,
    width: "42%",
    height: "60%",
    outline: "none",
    alignItems: "center",
    justifyContent: "center",
    display: "flex",
    flexDirection: "column",
    position: "absolute",
    top: "27%"
  },
  modalCard: {
    width: "92%",
    height: "90%",
    outline: "none",
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center",
    padding: 15
  },
  modalCloseButton: {
    backgroundColor: "#ffffff"
  },
  modalCardTitle: {
    width: "60%",
    marginBottom: "10%",
    fontSize: 24,
    fontWeight: "600",
    textAlign: "center",
    color: "#454545"
  },
  modalCardDetail: {
    fontSize: 18,
    fontWeight: "600",
    textAlign: "center",
    color: "#595959",
    marginTop: "10%"
  }
}));

const Transition = React.forwardRef(function Transition(props, ref) {
  return <Slide direction="up" ref={ref} {...props} />;
}); //comment for text editor 

const AccountProfile = props => {
  const { className, auth, modifyUserData, setLoader, isAvatarLoader, ...rest } = props;
  const classes = useStyles();

  const [displayavatarModal, setModalValue] = React.useState(false);
  const [oldFileList, setOldFileList] = React.useState([]);
  const [filePaths, setfilePaths] = React.useState([]);
  const [isLoading, setIsLoading] = React.useState(false);
  const [filesUploaded, setFilesUploaded] = React.useState([]);
  const [openConfirm, setOpenConfirm] = React.useState(false);

  const user = {
    name: auth.firstName+' '+auth.lastName,
    city: 'Los Angeles',
    country: 'USA',
    timezone: 'GTM-7',
    avatar: auth.avatar ? auth.avatar.name : '../../assets/images/alice.png'
  }

  const acceptedAvatarImg = "image/*";

  const openavatarModal = () => {
    setModalValue(true);
  }

  const handleClose = () =>{
    setModalValue(false);
    setOldFileList([]);
    setFilesUploaded([]);
    setOpenConfirm(false);
  }


  const uploadavatarFile = async (avatarfile) => {
    setLoader(true);
    setModalValue(false);
    const promise = await Promise.all(avatarfile.map(async (file) => {
      return await uploadFile(file);
    }));
    promise.map(path => avatarfile.push(path));

    let userData = {
      userID: auth.id,
      avatar: avatarfile[1]
    }
    modifyUserData(userData);

    setOldFileList([]);
    setFilesUploaded([]);
  }

  const onDrop = (uploadedFiles) => {
    setIsLoading(true);
    const files = _.uniqBy(uploadedFiles.flat(), function (e) {
      return JSON.stringify(e);
    });
    setOldFileList(files);
    setIsLoading(false)

    let fileNames;
    if (files && files.length >= 1) {
       fileNames = files.map((file, i) => (
        <li key={i} className={classes.list}>
          <span className={classes.listDot}></span>{file.name}
        </li>
      ));
      setFilesUploaded(fileNames);
    }
  }

  const handleRemoveAvatar = () =>{
    if(auth.existedAvatar){
      setOpenConfirm(true);
    }
  }

  const removeAvatar = async () => {
    if(auth.existedAvatar){
      setOpenConfirm(false);
      let userData = {
        userID: auth.id,
        avatar: null
      }
      modifyUserData(userData);
      await removeFileFromS3(auth.existedAvatar, 0);
    }
  }

  return (
    <Card
      {...rest}
      className={clsx(classes.root, className)}
    >
      <CardHeader
        title="User Information"
      />
        <Divider />

        {/* <CardContent> */}
          {/* <div className={classes.details}>
          { */}
            <div className={classes.details}>
                {isAvatarLoader ? 
                  <CircularProgress className={classes.avatar}
                    style={{ marginTop: 0, marginRight: 0 }}
                    size={10}
                    disableShrink
                  /> :
                  <Avatar
                    className={classes.avatar}
                    src={user.avatar}
                  />
                }
              <Typography >
                Name: {user.name}
              </Typography>
              <Typography>
                Email: {auth.email}
              </Typography>
            </div>
            {/* }
          </div> */}
        
          {/*<div className={classes.progress}>
            <Typography variant="body1">Profile Completeness: 70%</Typography>
            <LinearProgress
              value={70}
              variant="determinate"
            />
          </div> */}
        {/* </CardContent> */}

        <CardActions style={{ marginLeft: '0.5rem', marginBottom: '1rem' }}>
          <Button
            className={classes.uploadButton}
            color="primary"
            variant="contained"
            onClick={openavatarModal}
          >
            Upload Picture
          </Button>
          <Button className={classes.uploadButton} variant="contained"
            onClick={handleRemoveAvatar}
          >
            Remove Picture
          </Button>
        </CardActions>

        <Modal
          aria-labelledby="simple-modal-title"
          aria-describedby="simple-modal-description"
          open={displayavatarModal}
          onClose={handleClose}
          style={{ display: "flex", justifyContent: "center" }}
        >
          <div className={classes.modalContainer}>
            <Card className={classes.modalCard}>
              <Typography
                className={classes.modalCardTitle}
              >
                Upload your avatar
              </Typography>
              
              <Dropzone onDrop={onDrop} accept={acceptedAvatarImg}>
                {({ getRootProps, getInputProps }) => (
                  <section className={classes.container}>
                    <div {...getRootProps({ className: 'dropzone' })} style={{ height: "100%" }}>
                      <input {...getInputProps()} />
                      <div className={classes.dropzone} style={{textAlign: "center"}}>
                        <CloudDownloadIcon style={{ width: 80, height: 80 }} />
                      </div>
                    </div>
                  </section>
                )}
              </Dropzone>

              <aside style={{ maxHeight: 50, overflow: 'auto' }}>
                  {isLoading && (
                    <CircularProgress
                      style={{ marginTop: 10, marginLeft: 32, alignSelf: 'center' }}
                      size={20}
                      disableShrink
                    />
                  )}
                  {filesUploaded.length >= 1 && (
                    <div>
                      <ul style={{ marginLeft: 32, marginRight: 32, marginBottom: 10, marginTop: 10 }}>
                        {filesUploaded}
                      </ul>
                    </div>
                  )}
              </aside>
              {Boolean(oldFileList.length >= 1) && 
                <div style={{
                  display: "flex",
                  justifyContent: "flex-end",
                  height: 70,
                  alignItems: "center",
                  padding: 20
                }}>
                  <Button
                    variant="contained"
                    color="primary"
                    onClick={event=> uploadavatarFile(oldFileList)}
                  >
                    Upload
                  </Button>
                </div>
              }
            </Card>
            <div style={{
              justifyContent: "center",
              display: "flex",
              paddingTop: 5
            }}>
              <Fab
                variant="round"
                className={classes.modalCloseButton}
                onClick={handleClose}>
                <CloseIcon />
              </Fab>
            </div>
          </div>
        </Modal>

        <Dialog
          open={openConfirm}
          TransitionComponent={Transition}
          keepMounted
          aria-labelledby="alert-dialog-slide-title"
          aria-describedby="alert-dialog-slide-description"
        >
          <DialogTitle id="alert-dialog-slide-title">{"Please confirm"}</DialogTitle>
          <DialogContent>
            <DialogContentText id="alert-dialog-slide-description">
              Are you sure you want to remove this avatar image.
            </DialogContentText>
          </DialogContent>
            <DialogActions>
              <Button onClick={handleClose} color="primary">
                No
              </Button>
              <Button onClick={removeAvatar} color="primary">
                Yes
              </Button>
            </DialogActions>
        </Dialog>

    </Card>
         
  );
};

AccountProfile.propTypes = {
  className: PropTypes.string
};



export default AccountProfile;
/*export default connect(
  mapStateToProps
)(AccountProfile);*/
