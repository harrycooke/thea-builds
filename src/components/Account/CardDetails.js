import React from 'react';
import PropTypes from "prop-types";
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { 
  registerUser,
  setSnackbar,
  fetchPracticeData,
  fetchStripeData,
  modifyPracticeData,
  getPublicStripeKey,
  getSetupIntent
} from "../../actions/authActions";
import {
  Avatar,
  Button,
  Radio,
  RadioGroup,
  CardActions,
  CardContent,
  FormControlLabel,
  FormControl,
  FormLabel,
  CssBaseline,
  TextField,
  Link,
  Grid,
  Typography,
  Container,
  MenuItem,
  Fab,
  Card,
  CardHeader,
  Divider,
} from "@material-ui/core";
import { withStyles } from '@material-ui/core/styles';
import CloseIcon from "@material-ui/icons/Close";

import classnames from "classnames";
import { specialtyFullList } from "../../utils/constant";
import { getServerUrl } from "../../utils/functions";

import { loadStripe } from "@stripe/stripe-js";
import { Elements } from "@stripe/react-stripe-js";
import CardSetupForm from "../CardDetails/CardSetupForm";
import CircularProgress from '../SharedComponents/CircularProgress';





const styles = theme => ({
  root: {},
  details: {
    display: 'flex',
    flexDirection: "column",
    margin: "1.2em"
  },
  paper: {
    marginTop: theme.spacing(8),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    margin: "1.2em"

  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: theme.palette.secondary.main,
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(3),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
  errorText: {
    fontSize: 12,
    fontWeight: "500"
  },
});

class CardDetails extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      secret: "",
      name: "",
      lastFour: "",
      brand: "",
      updateCard: false,
      errors: {},
      stripeKey: "",
      paymentMethod: "",
      stripePromise: ""
    };
    
    this.handler = this.handler.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  

  UNSAFE_componentWillReceiveProps(nextProps) {
    if (nextProps.errors) {
      this.setState({
        errors: nextProps.errors
      });
    }
  }
  
  async componentDidMount() {
    const response = await this.props.fetchStripeData(this.props.auth.userdata.practice_id);
    const response2 = await this.props.fetchPracticeData(this.props.auth.userdata.practice_id);

    const { classes, auth } = this.props;

    if(this.props.auth.userdata.is_admin == true && !this.props.auth.practiceData.stripe_id){
      let practiceData = {
        practiceId: this.props.auth.userdata.practice_id,
        stripeId: this.props.auth.practiceData.stripe_id
      }
      await this.props.modifyPracticeData(practiceData); //get new stripe id
      //todo: set loading on element until this resolves (or at least disable add card button redux state should be preset)
    } 

    let key = await this.props.getPublicStripeKey();    
    let stripePromise = await loadStripe(key);

    this.setState({
      stripeKey: key,
      stripePromise: stripePromise
    });
  }

  handler(paymentMethod) {
    try{
      this.props.fetchStripeData(this.props.auth.userdata.practice_id);
      this.setState({
        updateCard: false,
      })

    } catch (error){
      console.log(error);
    }

  }


  onChange = e => {
    this.setState({ [e.target.id]: e.target.value });
  };

  UNSAFE_componentWillMount () {
    document.addEventListener('keydown', this.handleHitEnter, true)
    //this.setState({ updateCard: false });
    const { classes, auth } = this.props;
    this.props.fetchPracticeData(this.props.auth.userdata.practice_id);

    this.props.fetchStripeData(this.props.auth.userdata.practice_id);

  }
  
  componentWillUnmount() {
    document.removeEventListener('keydown', this.handleHitEnter, true)
  }
  
  handleHitEnter(e) {
    const ENTER_KEY_CODE = 13
    if (e.target.name === 'my-input' &&
       (e.key === 'Enter' || e.keyCode === ENTER_KEY_CODE)) {
        e.stopPropagation()
    }
  }
  
  handleChangeInput(e) {
    this.setState({ myInput: e.target.value })
  }

  canDisplayCard() {
    //may be better to just update state here
    //also must be a cleaner way to do this 
    if (this.props.auth.stripeData){
      if (this.props.auth.stripeData.data){
        if (this.props.auth.stripeData.data[0]){
          const cardData = this.props.auth.stripeData.data[0]
          if (cardData.billing_details && cardData.card){
            if (cardData.billing_details.name && cardData.card.last4 && cardData.card.brand){
              return true;
            }
          }
        }
      }
    } else {
      return false;
    }
  }


  async handleSubmit() {
    this.setState({ updateCard: true });
    
    const setupIntent = await this.props.getSetupIntent(this.props.auth.practiceData);
    this.setState({secret: setupIntent.client_secret})
  }

  render(){
    const { classes, auth } = this.props;
    const { errors, stripePromise } = this.state;

    const secret = this.state.secret;
    return (
    <div>
    <Card className={classes.root}>
        <CardHeader
          title="Payment Information"
        />
        <Divider />
        <CircularProgress isLoading={this.props.auth.loading} />
        { (this.state.updateCard)  &&
        (<div className={classes.details}>
          <Typography style={{ marginBottom: 20 }}>
             Please add your card details below.
          </Typography>
          <Grid container spacing={2}>
            <Grid item xs={12} sm={12}>
              <TextField
                autoComplete="name"
                name="name"
                variant="outlined"
                required
                fullWidth
                id="name"
                label="Name on card"
                autoFocus
                onChange={this.onChange}
                className={classnames("", {
                  invalid: errors.firstName
                })}
              />
              <Typography
                color="error"
                className={classes.errorText}
              >
                {errors.firstName}
              </Typography>
            </Grid>
          </Grid>
          <Elements stripe={stripePromise}>
           <CardSetupForm handler={this.handler} name={this.state.name} secret={secret} history={this.props.history}/>
          </Elements>
        </div>
        )}
        { (this.props.auth.practiceData.stripe_id && this.props.auth.stripeData && !this.props.auth.loading && !this.state.updateCard) &&       
        (<div>
          {this.canDisplayCard() ? (
          <div style={{ marginTop: '1rem', marginLeft: '1rem', marginBottom: '1rem', paddingBottom: '0' }}>
            <Typography variant="h6" component="h2">
              Card Details:
            </Typography>
            <Typography color="textSecondary">
              {this.props.auth.stripeData.data[0].billing_details.name}'s {this.props.auth.stripeData.data[0].card.brand}
            </Typography>
            <Typography>
              Card Number: **** {this.props.auth.stripeData.data[0].card.last4} 
            </Typography>
          </div>         
          ) : (
          <div style={{ marginTop: '1rem', marginLeft: '1rem', marginBottom: '1rem', paddingBottom: '0' }}>
            <Typography variant="h6" component="h2">
              No card on file. 
            </Typography>
          </div>    
        )} 

          <CardActions style={{ marginTop: '0rem', marginLeft: '0.5rem', marginBottom: '1rem' }}>
              <Button
                //marginLeft="3"
                color="primary"
                variant="contained"
                onClick={this.handleSubmit}
              >
              {this.canDisplayCard() ? 'Update' : 'Add'} Card
            </Button>
            </CardActions>
        </div>
        )}
    </Card>

    </div>

    );
  }
}

CardDetails.propTypes = {
  registerUser: PropTypes.func.isRequired,
  fetchPracticeData: PropTypes.func.isRequired,
  modifyPracticeData: PropTypes.func.isRequired,
  fetchStripeData: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired,
  getPublicStripeKey: PropTypes.func.isRequired,
  getSetupIntent: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  auth: state.auth,
  errors: state.errors
});

export default connect(
  mapStateToProps,
  { fetchPracticeData, modifyPracticeData, fetchStripeData, registerUser, setSnackbar, getPublicStripeKey, getSetupIntent }
)(withRouter(withStyles(styles)(CardDetails)));