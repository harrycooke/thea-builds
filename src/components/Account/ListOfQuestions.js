import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import classnames from "classnames";

import {
  Card,
  CardHeader,
  CardContent,
  CardActions,
  Divider,
  Slide,
  FormControlLabel,
  Switch,
  Grid,
  TextField,
  Button,
  Typography,
} from '@material-ui/core';
import { withStyles } from "@material-ui/styles";
import { modifyUserData, confirmPassword, logoutUser } from "../../actions/authActions";
import { sendAmplitudeData } from "../../utils/amplitude";
import { newAccountEmail, registerUser, setSnackbar } from "../../actions/authActions";        //newUserInviteEmail

const styles = theme => ({
  root: {},
  errorText: {
    fontSize: 12,
    fontWeight: "500"
  }
});

const Transition = React.forwardRef(function Transition(props, ref) {
  return <Slide direction="up" ref={ref} {...props} />;
  /// Comment to get rid of yellow on sublime
});

export class ListOfQuestions extends React.Component {
//   constructor(props) {
//     super(props);
//     this.state = {
//     name: '',
//     email: '',
//     text: '',
//     errors: {}
//     }
// }

  state = {
    name: '',
    email: '',
    text: '',
    errors: {},
    open: false,
    persons: []
  }

  handleText = i => e => {
    let persons = [...this.state.persons]
    persons[i] = e.target.value
    this.setState({
      persons
    })
  }
  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
};

  handleSubmit = () => {
    const { id } = this.props.auth;
    const { email, name } = this.state;
    console.log("submitting new user");
    this.props.newAccountEmail({        //newUserInviteEmail
        userID: id,
        inviteName: this.state.name,
        inviteEmail: this.state.email
    });

    if (name === '') {
        let errors = {
            name: "Name field is required."
        }
        this.setState({ errors })
      } else if (email === '') {
        let errors = {
            email: "Email field is required."
        }
        this.setState({ errors })
      }

    this.props.setSnackbar(true);
    this.setState({ open: true });
    //TODO: add success snackbar and error handling here 
}

  handleDelete = i => e => {
    e.preventDefault()
    let persons = [
      ...this.state.persons.slice(0, i),
      ...this.state.persons.slice(i + 1)
    ]
    this.setState({
      persons
    })
  }

  addQuestion = e => {
    e.preventDefault()
    let persons = this.state.persons.concat([''])
    this.setState({
      persons
    })
  }

  render() {
    const { classes } = this.props;
    const { modalOpened, open, errors, name, email } = this.state;

    return (
      <React.Fragment>
        <div className={classes.root}>
          {this.state.persons.map((person, index) => (
            <span key={index}>
              <Grid container spacing={3}>
                {/* <Grid item xs={6}>
                  <TextField
                    margin="normal"
                    type="name"
                    variant="outlined"
                    required
                    fullWidth
                    id="name" name="name"
                    style={{ marginLeft: 0, flexDirection: "row", display: "flex", justifyContent: "auto", width: "100%"}}
                    label="Provider Full Name"
                    onChange={this.handleText(index)}
                    value={person}
                  />
                  <TextField margin="normal" variant="outlined" required fullWidth id="name" name="name" type="name"
                    style={{ marginLeft: 0, flexDirection: "row", display: "flex", justifyContent: "auto", width: "100%"}}
                    label="Provider Full Name"
                    onChange={this.handleChange}
                    // className={classnames("", {
                    //     invalid: errors.name
                    //   })}
                    // value=""
                  />
                </Grid> */}

                <Grid item xs={6}>
                <TextField
                  margin="normal"
                  type="email"
                  variant="outlined"
                  required
                  fullWidth
                  id="email" name="email"
                  style={{ marginLeft: 0, flexDirection: "row", display: "flex", justifyContent: "auto", width: "100%"}}
                  label="Email Address"
                  onChange={this.handleText(index)}
                  // onChange={this.handleChange}
                  value={person}
                />
                {/* <TextField margin="normal" variant="outlined" required fullWidth id="email" name="email" type="email"
                  style={{ marginLeft: 0, flexDirection: "row", display: "flex", justifyContent: "auto", width: "100%"}}
                  label="Email Address"
                  onChange={this.handleChange}
                  // className={classnames("", {
                  //     invalid: errors.email
                  //   })}
                  // value=""
                /> */}
                </Grid>

                {/* <form >
                  <label>
                    Name:
                    <input type="text" value="" onChange={} />
                  </label>
                  <input type="submit" value="Submit" />
                </form> */}
              </Grid>
              
              <button onClick={this.handleDelete(index)}>X</button>
            </span>
          ))}


            <TextField multiline fullWidth rows={5} required rowsMax="10" margin="normal" variant="outlined" 
                defaultValue={"Hey friend, I wanted to invite you to try TheaHealth. It will allow me to securely send documents and images about patients who I would like to consult you on. Let me know if you choose to try it and I will send you a patient case!"}
                label="Email message"
                // value=""
                //onChange={this.handleQuestionChange}
                //className={classes.textField}
            />

          <Button 
            variant="contained" 
            color="inherit" backgroundColor="primary" 
            onClick={this.addQuestion}
            style={{ marginLeft: 0, flexDirection: "row", display: "flex", justifyContent: "auto"}}
            >
              <Typography >
                Add more
              </Typography>
          </Button>

          <Button 
            variant="contained" 
            // className={`${classes.sendButtonRightAlign} ${classes.addButton}`} 
            // disabled={!this.state.isLoading ? false : true} 
            color="primary" 
            backgroundColor="primary" 
            style={{ marginLeft: 0, flexDirection: "row", display: "flex", justifyContent: "auto", float: 'right'}} 
            onClick={this.handleSubmit} 
          >
            <Typography >
              Send
            </Typography>
          </Button>

        </div>
      </React.Fragment>
    )
  }
}

ListOfQuestions.propTypes = {
  auth: PropTypes.object.isRequired,
  newAccountEmail: PropTypes.func.isRequired,
  modifyUserData: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  auth: state.auth.userdata,
});

export default connect(
  mapStateToProps,
  { newAccountEmail, modifyUserData, setSnackbar}
)(withStyles(styles)(ListOfQuestions));
