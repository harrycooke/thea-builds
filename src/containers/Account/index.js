import React from 'react';
import { Redirect, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/styles';
import { 
  Grid,
  Button,
  CardActions,
  Snackbar,
  Card,
 } from '@material-ui/core';
import CreateIcon from '@material-ui/icons/Create';

//import AccountDetails from '../../components/Account/AccountDetails';
import AccountProfile from '../../components/Account/AccountProfile';
import NotificationSettings from '../../components/Account/NotificationSettings';
import ListOfQuestions from '../../components/Account/ListOfQuestions';
import ReferYourFriend from '../../components/Account/ReferYourFriend';
import Password from '../../components/Account/Password';
import UserCreation from '../../components/Account/UserCreation';
import CardDetails from '../../components/Account/CardDetails';
import FriendInviteTable from '../../components/Account/FriendInviteTable';
import MemberInviteTable from '../../components/Account/MemberInviteTable';

import { sendAmplitudeData } from '../../utils/amplitude';
import { modifyUserData, setSnackbar, setSnackbar2} from "../../actions/authActions";
import MySnackbarContentWrapper from '../../components/SharedComponents/SnackBar';
import MySnackbarContentWrapper2 from '../../components/SharedComponents/SnackBar2';

/*******
TODO:
- set number of invites in an account through practice db
*******/

const AdapterLink = React.forwardRef((props, ref) => <Link innerRef={ref} {...props} />);

const styles = theme => ({
  root: {
    height: "60%",
    display: "flex",
    marginRight: 0,
    marginLeft: 0,
    justifyContent: "center",        //flex-start, center, flex-end
    [theme.breakpoints.down('sm')]: {
      padding: theme.spacing(0),    
    },
    [theme.breakpoints.up('sm')]: {
      padding: theme.spacing(4),   
    },
  },
  uploadButton: {
    // width: "100%",
    marginRight: theme.spacing(2),
    [theme.breakpoints.down('sm')]: {
      fontSize: 12  
    },
    [theme.breakpoints.up('md')]: {
      fontSize: 15
    },
  },
});

/*
class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error) {
    // Update state so the next render will show the fallback UI.
    return { hasError: true };
  }

  componentDidCatch(error, errorInfo) {
    // You can also log the error to an error reporting service
//    logErrorToMyService(error, errorInfo);
    console.log(error);
    console.log(errorInfo)
  }

  render() {
    if (this.state.hasError) {
      // You can render any custom fallback UI
      return <h1>Something went wrong.</h1>;
    }
    return this.props.children; 
  }
}*/

class Account extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isAvatarLoader: false
    }
  }
  state = {
    open: true,
  };

  componentDidMount() {
    sendAmplitudeData('Open account setting page');
    
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    if(nextProps.auth.avatar && this.props.auth.avatar && nextProps.auth.avatar.name !== this.props.auth.avatar.name){
      this.setState({isAvatarLoader: false});
    }else if(nextProps.auth.avatar && this.props.auth.avatar == null && nextProps.auth.avatar.name){
      this.setState({isAvatarLoader: false});
    }
  }

  handleSnackbarClose = () => { 
    this.props.setSnackbar(false);
    // this.props.setSnackbar2(false);
  }
  handleEmailSnackbarClose = () => { 
    this.props.setSnackbar2(false);
  }


  setLoader = (value) => {
    this.setState({isAvatarLoader: value});
  }

  render() {
    const { classes, auth, modifyUserData } = this.props;
    const { open } = this.state;
    let props = {
      setLoader: this.setLoader,
      auth: auth,
      isAvatarLoader: this.state.isAvatarLoader,
      modifyUserData: modifyUserData
    };
    // console.log("authfsa");
    // console.log(auth);
    
    return (
      <div className={classes.root}>
        <div style={{
          display: "flex",
          flexDirection: "column",
          alignItems: "center",
          }}>
            <Grid
              container
              spacing={2}
              alignItems="center"
              style={{ display: "flex", justifyContent: "center" }}
            >{/*
              <Grid
                item
                lg={8}
                md={7}
                xl={9}
                xs={12}
              >
                <ReferYourFriend />
              </Grid>*/}
              {/* </Card> */}

              {/* <Grid
                item
                lg={8}
                md={7}
                xl={9}
                xs={12}
              >
                <ListOfQuestions />
              </Grid> */}

              {/* <Grid
                item
                lg={8}
                md={7}
                xl={9}
                xs={12}
              >
                <FriendInviteTable />
              </Grid> */}

              {/* <Grid
                item
                lg={8}
                md={7}
                xl={9}
                xs={12}
              >
                <MemberInviteTable />
              </Grid> */}

              <Grid
                item
                lg={8}
                md={7}
                xl={9}
                xs={12}
              >
                <NotificationSettings />
              </Grid>

              <Grid
                item
                lg={8}
                md={7}
                xl={9}
                xs={12}
              >
                <Password />
              </Grid>

              { auth.is_admin && ( 
                <Grid
                  item
                  lg={8}
                  md={7}
                  xl={9}
                  xs={12}
                >
                  
                  <CardDetails {...props}/>
                  
                </Grid>
                )
              }

              <Grid
                item
                lg={8}
                md={7}
                xl={9}
                xs={12}
              >
                <AccountProfile {...props}/>
              </Grid>
              
              {/* <Grid
                item
                lg={8}
                md={7}
                xl={9}
                xs={12}
              >
                <Button
                  className={classes.uploadButton}
                  fullWidth
                  color="primary"
                  variant="contained"
                  component={AdapterLink}
                  to="/pricing"
                >
                  Pricing - Change plan
                </Button>
              </Grid> */}

              { /*auth.is_admin && ( 
                <Grid
                  item
                  lg={8}
                  md={7}
                  xl={9}
                  xs={12}
                >
                  <UserCreation {...props}/>
                </Grid>               
                )
              */} 

            </Grid>
        </div>
        <Snackbar
            anchorOrigin={{
              vertical: 'bottom',
              horizontal: 'left',
            }}
            open={this.props.showSnackbar}
            autoHideDuration={3000}
            onClose={this.handleSnackbarClose}
          >
            <MySnackbarContentWrapper
              onClose={this.handleSnackbarClose}
              variant="success"
              message="Payment method added."
            />
        </Snackbar>      
        <Snackbar
            anchorOrigin={{
              vertical: 'top',
              horizontal: 'center',
            }}
            open={this.props.showSnackbar2}
            autoHideDuration={5000}
            onClose={this.handleEmailSnackbarClose}
          >
            <MySnackbarContentWrapper2
              onClose={this.handleEmailSnackbarClose}
              variant="success"
              message="Email has been sent. Please make sure your friend checks their spam folder in case they don't see the email."
            />
        </Snackbar>        
      </div>
    );
  }
};

Account.propTypes = {
  auth: PropTypes.object.isRequired,
  modifyUserData: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  auth: state.auth.userdata,
  showSnackbar: state.auth.showSnackbar,
  showSnackbar2: state.auth.showSnackbar2,
});


export default connect(
  mapStateToProps,
  { modifyUserData, setSnackbar, setSnackbar2 }
)(withStyles(styles)(Account));
