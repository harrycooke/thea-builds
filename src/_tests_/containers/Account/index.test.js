import React from 'react';
import ReactDOM from'react-dom';
import TestRenderer from 'react-test-renderer'; // ES6
import sinon from 'sinon';
import { mount } from 'enzyme';

import {Account} from '../../../containers/Account/index';

describe('Account', () => {
    beforeAll = () => {
        this.state = {
            isAvatarLoader: false
          }
    };

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(Account, div);
    });

    it('renders React components, traversing a ReactElement tree and represents as JSON object', () => {
        const renderer = TestRenderer.create(Account);
        renderer.toJSON();
      });
    

    describe('Account', () => {
        it('should render correctly in "debug" mode', () => {
          const component = (Account);
          expect(component).toMatchSnapshot();
        });
      });

    // it('calls funcName handleSnackbarClose', () => {
    //     sinon.spy(Account.prototype, 'handleSnackbarClose');
    //     const wrapper = mount(<Account />);
    //     expect(Account.prototype.componentDidMount).to.have.property('callCount', 1);
    //   });
    
    //   it('allows us to set props', () => {
    //     const wrapper = mount(<Account bar="baz" />);
    //     expect(wrapper.props().bar).to.equal('baz');
    //     wrapper.setProps({ bar: 'foo' });
    //     expect(wrapper.props().bar).to.equal('foo');
    //   });
    
    //   it('simulates click events', () => {
    //     const onButtonClick = sinon.spy();
    //     const wrapper = mount((
    //       <Account onButtonClick={onButtonClick} />
    //     ));
    //     wrapper.find('button').simulate('click');
    //     expect(onButtonClick).to.have.property('callCount', 1);
    //   });
});

test('Fake test', ()=>{
    expect(true).toBeTruthy();          //here toBeTruthy is a predifined method whichif receives "true", considers to be expected.
});