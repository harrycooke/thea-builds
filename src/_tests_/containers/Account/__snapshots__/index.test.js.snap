// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Account should render correctly in "debug" mode 1`] = `undefined`;
